lib/membrane/rtp/outbound_packet_tracker.ex

```diff
@@ -91,17 +122,28 @@ defmodule Membrane.RTP.OutboundPacketTracker do
 
     buffer = %Buffer{payload: payload, metadata: metadata}
 
-    {{:ok, buffer: {:output, buffer}}, state}
+    {{:ok, buffer: {:output, buffer}}, %{state | any_buffer_sent?: true}}
+  end
+
+  @impl true
+  def handle_other(:send_stats, _ctx, %{rtcp_output_pad: nil} = state) do
+    {:ok, state}
   end
 
   @impl true
   def handle_other(:send_stats, _ctx, state) do
     stats = get_stats(state)
-    state = %{state | any_buffer_sent?: false}
-    {{:ok, notify: {:outband_stats, stats}}, state}
+
+    actions =
+      %{state.ssrc => stats}
+      |> SenderReport.generate_report()
+      |> Enum.map(&Membrane.RTCP.Packet.serialize(&1))
+      |> Enum.map(&{:buffer, {state.rtcp_output_pad, %Membrane.Buffer{payload: &1}}})
+
+    {{:ok, actions ++ [redemand: state.rtcp_output_pad]}, %{state | any_buffer_sent?: false}}
   end
 
-  @spec get_stats(State.t()) :: map() | :no_stats
+  @spec get_stats(State.t()) :: SenderReport.sender_stats_t() | :no_stats
```

Comment:
Actually, it seems it's one entry of sender_stats_t. Check if dialyzer passes

Reply:
It is a private function in the end, I don't know why I put the spec there in the first place...

Reply:
It improves readability and allows dialyzer to check types (as long as the types are correct :P), but it's optional indeed